hyperspy/signal.py

```diff
@@ -4989,26 +4994,27 @@ def _map_iterate(
                 sig.data = sig.data.compute(num_workers=max_workers)
         return sig
 
-    def _get_drop_axis_new_axis(self, output_signal_size):
-        am = self.axes_manager
-        if output_signal_size == self.axes_manager.signal_shape:
-            drop_axis = None
-            new_axis = None
-            axes_changed = False
-        else:
-            axes_changed = True
-            if len(output_signal_size) != len(am.signal_shape):
-                drop_axis = am.signal_indices_in_array
-                nav_dim = am.navigation_dimension
-                new_axis = tuple(range(nav_dim, len(output_signal_size) + nav_dim))
-            else:
-                drop_axis = [it for (o, i, it) in zip(output_signal_size,
-                                                      am.signal_shape,
-                                                      am.signal_indices_in_array)
-                             if o != i]
-                drop_axis = tuple(drop_axis)
-                new_axis = drop_axis
-        return drop_axis, new_axis, axes_changed
+    def get_block_pattern(self, args, output_shape=None):
```

Comment:
Can `output_shape` ever be `None`? If not, `output_shape=None` should not be used.

```python
output_shape = None
len(output_shape) # TypeError: object of type 'NoneType' has no len()
```